### PR TITLE
Linting rules

### LINKED ISSUE #13: Linting rules
We need to provide a basic linting rule and style guide. The linting rules should be defined by a configuration file which is included in the template

### PR DESCRIPTION
**Pull request by aiola**
Resolves #13

### CODE REVIEW

File: .mypy.ini [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+ignore_missing_imports = True
+
+[mypy-sklearn.*]
+ignore_missing_imports = True
+
+[mypy-seaborn.*]
+ignore_missing_imports = True
+
+[mypy-scipy.*]
+ignore_missing_imports = True
```
Comment (deusebio) [inline .mypy.ini:20]: Bellissimo! But does this mean that we can skip the " # type: ignore" everywhere in the code (v. cgnal-core)?

**Review by deusebio — APPROVED** [review]
Looks good!

**Review by NicolaDonelli — APPROVED** [review]
Wouldn't it be possible to move all the configuration files in the confs folder?
Are there checks on presence of docstrings on every object?

Except for that minor detail, I am fine with the content.

File: .flake8 [hunk 2]
```diff
@@ -0,0 +1,7 @@
+[flake8]
+ignore = E203, E266, E501, W503
+# line length is intentionally set to 80 here because black uses Bugbear
+# See https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html#line-length for more details
+max-line-length = 80
+max-complexity = 18
+select = B,C,E,F,W,T4,B9
```
Comment (NicolaDonelli) [inline .flake8:7]: It would be better to include this section in the setup.cfg file

**Review by NicolaDonelli — CHANGES_REQUESTED** [review]

File: .mypy.ini (hunk 1 above)
Comment (NicolaDonelli) [inline .mypy.ini:20]: Couldn't this file be included as a section in the `setup.cfg` file?

**Review by NicolaDonelli — APPROVED** [review]

**Pull request merged**